lib/batch/add_publisher_for_collection_members.rb

```diff
@@ -0,0 +1,18 @@
+# updates all the items in a collection with no publisher value.
+# Requires a collection_id
+
+def add_publisher_for_collection_members(collection_id:)
+  collection = Collection.find(collection_id)
+
+  puts "Checking #{collection.members.count} items in #{collection.title} for empty publisher"
+  collection.members.each do |member|
+    if member.class == Collection
+      add_publisher_for_collection_members(collection_id: member.id)
```

Comment:
Interesting recursion here. Just so I'm clear , when `member` is a `Collection` is that representing when a `member` owns a group of things?

Reply:
Ah, after reading the spec I think this is meant to handle nested Collections. Seemingly that's when a Collection has a Collection within it right?

Reply:
Yes, exactly that. It's possible Collections can have other Collections, in addition to GenericFiles.

Reply:
Got it. Yeah I think that's a testament to the quality of your tests and code that I was able to deduce that. Good stuff Austin.